### ISSUE TITLE
Create an initial file checklist for honeyclient integrity checks

### ISSUE THREAD

**Issue by dkindlund** [dkindlund/honeyclient#42]
Currently, in the honeyclient.xml file, there's no initial checklist for the honeyclient integrity checks to start with. Adding this checklist will help to reduce the false positives with the integrity checks.

**Comment by dkindlund** [comment]
Author: kindlund
Created bug branch for starters.

**Comment by dkindlund** [comment]
Author: kindlund
While trying to resolve this issue, we ran into fundamental issues with the current registry parsing code.  As a result, we rebuilt the entire registry subsystem, using some of the lessons learned from Xeno's first implementation.

Two new modules were added:

HoneyClient::Agent::Integrity::Registry[[BR]]
HoneyClient::Agent::Integrity::Registry::Parser

The registry parsing code is fast; however, the process of dumping each hive out to disk remains relatively slow (~ 1-6 mins per hive).

I retrofitted Xeno's original patch to include the <file_checklist> parameter within the new implementation of Integrity.  Users can specify which directories the Integrity module should explicitly monitor, by editing the etc/file_checklist.txt file.

Currently, in order to add/modify the corresponding registry excludes, the user will need to edit the Registry.pm file, manually.  We can include file list support in a future version of the code.

So far, the only downside to using the new parser implementation, is that Parse::Yapp is extremely sensitive to other libraries that make use of the notorious $', $`, and $& variables (as documented in the ParsingRegistry wiki page).  This doesn't mean existing libraries need to be abandoned; however, it does mean that some applicable libraries may need to be slightly patched (i.e., XML::XPath).

As it stands, this code is ready for testing inside our VM test environment.  I suspect some additional modules will need to be updated/installed, to get this new version to work properly.
